plugins/modules/device_configs_backup_workflow_manager.py

```diff
@@ -514,109 +523,101 @@ def validate_site_exists(self, site_name):
-                                msg = "Skipping device {0} in site {1} as its status is {2} or collectionStatus is {3} ".format(
-                                    item_dict["managementIpAddress"], site_name, item_dict["reachabilityStatus"], item_dict["collectionStatus"])
-                                self.log(msg, "WARNING")
             else:
-                # If unable to retrieve device information, log an error message
-                self.log("No response received from API call 'get_membership' to get membership information for site. {0}".format(site_name), "ERROR")
+                response = self.dnac._exec(
+                    family="site_design",
+                    function="get_site_assigned_network_devices",
+                    op_modifies=True,
+                    params=site_params,
+                )
 
-            # Log the total number of devices processed and skipped
-            self.log("Total number of devices processed: {0}".format(processed_device_count), "INFO")
-            self.log("Number of devices skipped due to being unreachable or APs: {0}".format(skipped_device_count), "INFO")
-            self.log("Filtered devices available for configuration backup: {0}".format(mgmt_ip_to_instance_id_map), "INFO")
+                self.log("Received API response from 'get_site_assigned_network_devices': {0}".format(str(response)), "DEBUG")
+                for device in response.get("response", []):
+                    device_id = device.get("deviceId")
+                    if device_id:
+                        device_response = self.dnac._exec(
+                            family="devices",
+                            function="get_device_by_id",
+                            op_modifies=True,
+                            params={"id": device_id}
+                        )
+
+                        management_ip = device_response.get("response", {}).get("managementIpAddress")
+                        if management_ip:
+                            mgmt_ip_to_instance_id_map[management_ip] = device_id
+                        else:
+                            self.log(f"Management IP not found for device ID: {device_id}", "WARNING")
 
         except Exception as e:
-            # Log an error message if any exception occurs during the process
-            self.log("Unable to fetch the device(s) associated to the site '{0}' due to {1}".format(site_name, str(e)), "ERROR")
+            self.log("Unable to fetch the device(s) associated with the site '{0}' due to {1}".format(site_name, str(e)), "ERROR")
 
-        # Log a warning if no reachable devices are found
         if not mgmt_ip_to_instance_id_map:
-            self.log("Reachable devices not found at Site: {0}".format(site_name), "WARNING")
+            self.msg = "No reachable devices found at Site: {0}".format(site_name)
+            self.update_result("ok", False, self.msg, "INFO")
+            self.module.exit_json(**self.result)
```

Comment:
Why do we need to exit from the module if devices not found at site?

Reply:
If the device ID is not available, then the IP address cannot be retrieved in this task